ecosystem/sep-0024.md

```diff
@@ -0,0 +1,688 @@
+
+**Note:** Both wallets and anchors should implement a sandbox mode for testing that uses the Stellar testnet and fake banking rails so counterparties can run through the flow without a need to collaborate.
+
+### Basic Wallet Implementation
+
+* Identify anchors you want to support manually, and test them with your wallet to be sure they work before whitelisting them. We encourage you to support as many anchors as possible.
+* For each anchor, use information from its [`stellar.toml`](sep-0001.md) file and its [`/info`](#info) endpoint to display useful information to the user about the asset they've picked.
+* Provide a UI that allows users to pick an asset, anchor, and amount to use for deposit or withdraw. The UI should display the asset's fee structure (if possible) as well as information such as the address of the anchor and description of the asset from the `stellar.toml` file.
+* **Use the `/info` endpoint**
+    * Fetch the asset's deposit & withdawal fee structure: if `fee_fixed` and `fee_percent` are provided, show this to the user early in the process so they're fully informed.
+    * If the `/fee` endpoint is enabled, use it for computing fees when you need to show them to the user.
+* **Authentication**
+   * Perform [authentication](#authentication) via SEP-10 before hitting those endpoints
+* **Make a request to `/deposit` or `/withdraw`.**
+    * This will respond with the interactive url needed to proceed with KYC and deposit/withdraw details.
+* **For `/deposit` and `/withdraw`**
+    * Optionally attach any fields from [SEP-9](sep-0009.md) as query parameters in the interactive URL, in order to let the anchor pre-fill them in the interactive flow UI.  This is optional, but can create a much nicer user experience.
+    * If you do attach fields, make sure you attach them using a proper URL builder instead of just appending a string, as the interactive URL may already have query parameters or a hash-fragment value.
+* **For `/deposit`**
+    * Handle the interactive flow, handle it as [described in detail](#3-customer-information-needed-interactive).
+    * Handle the [special cases](#special-cases), they're relatively common.
+* **For `/withdraw`**
+    * Handle the interactive flow, handle it as [described in detail](#3-customer-information-needed-interactive).
+    * Do not handle the non-interactive or status responses to the `/withdraw` request.  These may be from the old [SEP6](sep-0006.md) protocol.
+    * Some wallets might exchange currencies only once they're ready to send the withdrawal payment, so exchange rate fluctuations might require withdrawal values to slightly vary from the originally provided `amount`. Anchors are instructed to accept a variation of ±10% between the informed `amount` and the actual value sent to the anchor's Stellar account. The withdrawn amount will be adjusted accordingly.
```

Comment:
> Is this because a user might have TOK, but want to withdraw USD, and the app won't convert the TOK=>USD until the withdrawal is confirmed to be going ahead?

Yes, that sounds right.

> Is the assumption that this is okay because what the anchor will withdraw will be one-to-one with the token the wallet is sending? How will this work in cases where it isn't one-to-one, or the rate varies with the amount?

Yes. If what the user is sending is less, then the anchor will withdraw less. The fees will be applied using the amount the user transfers to the anchor as a reference. The wallet could use the `/fee` or `/info` endpoints for double checking, in case there is any variation. In any case, we're adding this max ±10% margin to minimize any of these side effects.

Also, this should help providing more context: #395